src/main/kotlin/com/ampnet/mailservice/service/impl/FileServiceImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.ampnet.mailservice.service.impl
+
+import com.ampnet.mailservice.exception.ResourceNotFoundException
+import com.ampnet.mailservice.service.FileService
+import org.springframework.core.io.Resource
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpMethod
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.exchange
+import java.io.IOException
+import java.io.InputStream
+
+@Service
+class FileServiceImpl(private val restTemplate: RestTemplate) : FileService {
```

Comment:
It would be nice to store downloaded files because most of the time the same document will be used.